pkg/util/tls/tls.go

```diff
@@ -13,42 +13,65 @@ import (
 
 // ClientConfig is the config for client TLS.
 type ClientConfig struct {
-	CertPath string `yaml:"tls_cert_path"`
-	KeyPath  string `yaml:"tls_key_path"`
-	CAPath   string `yaml:"tls_ca_path"`
+	CertPath           string `yaml:"tls_cert_path"`
+	KeyPath            string `yaml:"tls_key_path"`
+	CAPath             string `yaml:"tls_ca_path"`
+	InsecureSkipVerify bool   `yaml:"tls_insecure_skip_verify"`
 }
 
+var (
+	errKeyMissing  = errors.New("certificate given but no key configured")
+	errCertMissing = errors.New("key given but no certificate configured")
+)
+
 // RegisterFlagsWithPrefix registers flags with prefix.
 func (cfg *ClientConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
-	f.StringVar(&cfg.CertPath, prefix+".tls-cert-path", "", "TLS cert path for the client")
-	f.StringVar(&cfg.KeyPath, prefix+".tls-key-path", "", "TLS key path for the client")
-	f.StringVar(&cfg.CAPath, prefix+".tls-ca-path", "", "TLS CA path for the client")
+	f.StringVar(&cfg.CertPath, prefix+".tls-cert-path", "", "Path to the client certificate file, which will be used for authenticating with the server. Also requires the key path to be configured.")
+	f.StringVar(&cfg.KeyPath, prefix+".tls-key-path", "", "Path to the key file for the client certificate. Also requires the client certificate to be configured.")
+	f.StringVar(&cfg.CAPath, prefix+".tls-ca-path", "", "Path to the CA certificates file to validate server certificate against. If not set, the host's root CA certificates are used.")
+	f.BoolVar(&cfg.InsecureSkipVerify, prefix+".tls-insecure-skip-verify", false, "Skip validating server certificate.")
 }
 
 // GetTLSConfig initialises tls.Config from config options
 func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
-	if cfg.CertPath != "" && cfg.KeyPath != "" && cfg.CAPath != "" {
-		clientCert, err := tls.LoadX509KeyPair(cfg.CertPath, cfg.KeyPath)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to load TLS certificate %s,%s", cfg.CertPath, cfg.KeyPath)
-		}
+	// no tls config given at all
+	if cfg.CertPath == "" && cfg.KeyPath == "" && cfg.CAPath == "" && !cfg.InsecureSkipVerify {
```

Comment:
Since we're focusing on the bugfix only in this PR, TLS should not be enabled if no certs are provided but `cfg.InsecureSkipVerify` is `true`, so:

```suggestion
	if cfg.CertPath == "" && cfg.KeyPath == "" && cfg.CAPath == "" {
```

Reply:
Very good catch. I wanted to get rid of this, but was to quick pulling the features out